.github/monitor-typescript-errors/utils/xml.js

```diff
@@ -0,0 +1,66 @@
+const { XMLParser } = require( 'fast-xml-parser' );
+
+exports.parseXml = ( filePath ) => {
+	const parser = new XMLParser( {
+		ignoreAttributes: false,
+		attributeNamePrefix: '',
+		attributesGroupName: '',
+	} );
+	const parsedFile = parser.parse( filePath );
+
+	return getDataFromParsedXml( parsedFile );
+};
+
+const getErrorInfo = ( error ) => {
+	const line = error.line;
+	const column = error.column;
+	const message = error.message;
+
+	return `${ line }:${ column } - ${ message }`;
+};
+
+const getDataFromParsedXml = ( parsedXml ) => {
+	const data = parsedXml.checkstyle.file;
+
+	return data.reduce(
+		( acc, { name, error } ) => {
+			const pathFile = name;
+			const hasMultipleErrors = Array.isArray( error );
+
+			return {
+				files: {
+					[ pathFile ]: hasMultipleErrors
+						? error.map( getErrorInfo )
+						: [ getErrorInfo( error ) ],
+					...acc.files,
+				},
+				totalErrors:
+					acc.totalErrors + ( hasMultipleErrors ? error.length : 1 ),
+				totalFilesWithErrors: acc.totalFilesWithErrors + 1,
+			};
+		},
+		{
+			totalErrors: 0,
+			totalFilesWithErrors: 0,
+		}
+	);
+};
+
+exports.getFilesWithNewErrors = (
+	newCheckStyleFileParsed,
+	currentCheckStyleFileParsed
+) => {
+	const newFilesReport = newCheckStyleFileParsed.files;
+	const currentFilesReport = currentCheckStyleFileParsed.files;
+
+	return Object.keys( newFilesReport ).reduce(
+		( acc, pathfile ) =>
+			currentFilesReport[ pathfile ] === undefined ||
```

Comment:
Should we intentionally compare the type, here? Just in case somewhere in the code `undefined` was assigned to a value.

```suggestion
			typeof currentFilesReport[ pathfile ] === 'undefined' ||
```

Reply:
> Should we intentionally compare the type, here? Just in case somewhere in the code, `undefined` was assigned to a value.
It is impossible that undefined was assigned to a value because everything is generated by the `@bartekbp/typescript-checkstyle` package.

I prefer to keep this version for readability reasons, but if you think the change is necessary, I will do it.

Reply:
Personally, I don't find this more readable (first time I saw this type of check being done without using `typeof` 😉), and although chances are minimal, `undefined` [is not a reserved word and it is writable outside global scope](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined).